Call completion handlers if delegate doesn't implement methods

iainx opened the pull request:
The completion handlers still need called if the user's delegate doesn't implement them, or URLSession will have an API Misuse error

iainx committed:
Call completion handlers if delegate doesn't implement methods

The completion handlers still need called if the user's delegate doesn't implement them, or URLSession will have an API Misuse error

iainx commented:
Sorry, forgot this needed checked, will try and add a test for it in the morning

iainx committed:
Add some extra tests to check delegate methods

Check the delegate methods we override in the chunks method don't give errors

iainx commented:
Added some more tests that check the delegate methods we override return correctly if the user doesn't supply their own delegate

mattmassicotte commented:
Huh, only on watchOS:

> [Tests/PacketTests/URLSessionTests.swift#L85](https://github.com/mattmassicotte/Packet/pull/3/files#annotation_20679265992)
> testErrorDelegate, XCTAssertEqual failed: ("NSURLErrorDomain") is not equal to ("PacketTests.MockError")

What do you make of that?

iainx committed:
Don't check which error is caught

URLSession seems to change the error type that is thrown during an error so it's hard to check which type of error was thrown, but really we just want to test that errors can be thrown
